### PR TITLE
Begin Azure Documentation section

### PR DESCRIPTION
**Pull request by stefangordon**
This gets some Azure documentation established, which will make it easier for others to start contributing to it.

There are some useful notes on how to authenticate, and also some simple sample policies.  I'm happy to move/restructure this as needed based on feedback.

Docs build here: http://pgreen.blob.core.windows.net/docbuild/index.html

image

### CODE REVIEW

File: docs/source/azure/authentication.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      "password": password,
+      "tenant": guid
+    }
+
+You will need to map it to environment variables for Custodian like this:
+
+.. code-block:: bash
+
+    AZURE_TENANT_ID=tenant
+    AZURE_SUBSCRIPTION_ID=subscription id
```
Comment (aluong) [inline docs/source/azure/authentication.rst]: (minor) for consistency of naming "subscriptionId"

File: docs/source/azure/authentication.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    AZURE_CLIENT_ID=appId
+    AZURE_CLIENT_SECRET=password
+
+
+
+Azure CLI
+---------
+
+Set environment variable :code:`AZURE_CLI_AUTH` to any value, and session will pull credentials and default
```
Comment (aluong) [inline docs/source/azure/authentication.rst]: (minor) "and the default"

File: docs/source/azure/contribute.rst [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+The c7n developer install includes c7n_azure.  A shortcut for creating a virtual env for development is available
+in the makefile:
+
+.. code-block:: bash
+
+    $ make install
+    $ source ../cc_env/bin/activate
+
+This creates a virtual env above your enlistment and installs all packages as editable.
```
Comment (aluong) [inline docs/source/azure/contribute.rst]: "This creates a virtual env in your enlistment..."

**Comment by kapilt** [comment]
lgtm, thanks

**Comment by kapilt** [comment]
also note on commit messages, we tend to a particular format, ie this would be `docs - begin azure documentation section`

**Pull request merged**